### PR TITLE
Added Repository pattern

### PR DESCRIPTION
**Pull request by dotnetdew**

### CODE REVIEW

File: Areas/Admin/ParkingZoneController.cs [hunk 1]
```diff
@@ -55,42 +54,43 @@ public IActionResult Create()
...
 
-            var parkingZone = await _context.ParkingZones.FindAsync(id);
+            var parkingZone = _repository.GetById(id);
+
             if (parkingZone == null)
-            {
                 return NotFound();
-            }
+
+            _repository.Update(parkingZone);
```
Comment (napster1991) [inline Areas/Admin/ParkingZoneController.cs:81]: Why updating?

File: Areas/Admin/ParkingZoneController.cs [hunk 2]
```diff
@@ -55,42 +54,43 @@ public IActionResult Create()
...
+        //POST: Admin/ParkingZones/Edit/5
         // To protect from overposting attacks, enable the specific properties you want to bind to.
         // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public async Task<IActionResult> Edit(Guid id, ParkingZone parkingZone)
+        public IActionResult Edit(Guid id, ParkingZone parkingZone)
         {
-            if (id != parkingZone.Id)
+            if (id == null)
```
Comment (napster1991) [inline Areas/Admin/ParkingZoneController.cs:93]: Revert  this change

File: Program.cs [hunk 3]
```diff
@@ -10,6 +12,8 @@
     options.UseSqlServer(connectionString));
 builder.Services.AddDatabaseDeveloperPageExceptionFilter();
 
+builder.Services.AddScoped(typeof(IParkingZoneRepository), typeof(ParkingZoneRepository));
```
Comment (napster1991) [inline Program.cs:15]: builder.Services.AddScoped<IParkingZoneRepository, ParkingZoneRepository>();

File: Repositories/IGenericRepository.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Parking_Zone.Repositories
+{
+    public interface IGenericRepository<T> where T : class
```
Comment (napster1991) [inline Repositories/IGenericRepository.cs:3]: Recommended name: just IRepository

File: Areas/Admin/ParkingZoneController.cs (hunk 2 above)
Comment (dotnetdew) [reply Areas/Admin/ParkingZoneController.cs:93]: reverted

File: Repositories/GenericRepository.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
+﻿
+using Microsoft.EntityFrameworkCore;
+using Parking_Zone.Data;
+
+namespace Parking_Zone.Repositories
+{
+    public class GenericRepository<T> : IGenericRepository<T> where T : class
```
[the remark this answers is not in the document]
Comment (dotnetdew) [reply Repositories/GenericRepository.cs:7]: changed to Repository

**Pull request merged**